packages/expo-asset/src/Asset.ts

```diff
@@ -24,6 +27,8 @@ type DownloadPromiseCallbacks = {
 
 export type AssetMetadata = AssetSources.AssetMetadata;
 
+const MANAGED_ENV = !!Constants.manifest;
```

Comment:
not the best way to say "if this is an Expo environment (client, standalone, expokit)" probably

Reply:
In other places `Constants.appOwnership` is used:

https://github.com/expo/expo/blob/ef951f1400e2d2428a033985b891c917b7820adb/packages/expo/src/Linking/Linking.ts#L9

https://github.com/expo/expo/blob/ef951f1400e2d2428a033985b891c917b7820adb/packages/expo/src/Linking/Linking.ts#L44-L46

Maybe it would suite this situation well too? It may return `expo` (in Expo Client)`, `standalone` (in ejected or Turtle apps), `guest` (_opened through a link from a standalone app_ as per the docs) or `undefined` in vanilla RN apps.